iot/api-client/beta-features/commands/receive/receive.py

```diff
@@ -0,0 +1,229 @@
+    # Register message callbacks. https://eclipse.org/paho/clients/python/docs/
+    # describes additional callbacks that Paho supports. In this example, the
+    # callbacks just print to standard out.
+    client.on_connect = on_connect
+    client.on_publish = on_publish
+    client.on_disconnect = on_disconnect
+    client.on_message = on_message
+
+    # Connect to the Google MQTT bridge.
+    print('Connecting with id: {} and pass: {}'.format(client_id, password))
+    client.connect(mqtt_bridge_hostname, mqtt_bridge_port)
+
+    # This is the topic that the device will receive configuration updates on.
+    mqtt_command_topic = '/devices/{}/commands/#'.format(device_id)
+
+    # Subscribe to the config topic.
+    print('Subscribing to {}'.format(mqtt_command_topic))
+    client.subscribe(mqtt_command_topic, qos=1)
+
+    return client
+# [END iot_mqtt_config]
+
+
+def parse_command_line_args():
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(description=(
+            'Example Google Cloud IoT Core MQTT device connection code.'))
+    parser.add_argument(
+            '--project_id',
+            default=os.environ.get('GOOGLE_CLOUD_PROJECT'),
+            help='GCP cloud project name')
+    parser.add_argument(
+            '--registry_id', required=True, help='Cloud IoT Core registry id')
+    parser.add_argument(
+            '--device_id', required=True, help='Cloud IoT Core device id')
+    parser.add_argument(
+            '--private_key_file',
+            required=True, help='Path to private key file.')
+    parser.add_argument(
+            '--algorithm',
+            choices=('RS256', 'ES256'),
+            required=True,
+            help='Which encryption algorithm to use to generate the JWT.')
+    parser.add_argument(
+            '--cloud_region', default='us-central1', help='GCP cloud region')
+    parser.add_argument(
+            '--ca_certs',
+            default='roots.pem',
+            help=('CA root from https://pki.google.com/roots.pem'))
+    parser.add_argument(
+            '--mqtt_bridge_hostname',
+            default='mqtt.googleapis.com',
+            help='MQTT bridge hostname.')
+    parser.add_argument(
+            '--mqtt_bridge_port',
+            choices=(8883, 443),
+            default=8883,
+            type=int,
+            help='MQTT bridge port.')
+    parser.add_argument(
+            '--jwt_expires_minutes',
+            default=20,
+            type=int,
+            help=('Expiration time, in minutes, for JWT tokens.'))
+
+    return parser.parse_args()
+
+
+# [START iot_mqtt_run]
+def main():
+    global minimum_backoff_time
+
+    args = parse_command_line_args()
+
+    # Add any JWT refresh logic here
+    client = get_client(
+        args.project_id, args.cloud_region, args.registry_id, args.device_id,
+        args.private_key_file, args.algorithm, args.ca_certs,
+        args.mqtt_bridge_hostname, args.mqtt_bridge_port)
+
+    for i in range(1, 120):
```

Comment:
Maybe a comment that this runs for 2 minutes and stops and that in production you'll probably want a "while True" loop?

Reply:
Done.